pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/MetsTransformerWorkerService.scala

```diff
@@ -41,25 +43,41 @@ class MetsTransformerWorkerService(
     })
   }
 
-  private def process(key: Version[String, Int]) = {
+  private def process(key: Version[String, Int]) =
     for {
-      metsData <- getMetsData(key)
-      metsString <- getFromMetsStore(metsData)
-      mets <- MetsXmlParser(metsString)
+      metsLocation <- getMetsLocation(key)
+      metsXml <- getMetsXml(metsLocation)
+      mets <- metsXml.toMetsData
       work <- mets.toWork(key.version)
       _ <- messageSender.sendT(work).toEither
     } yield ()
-  }
 
-  private def getMetsData(key: Version[String, Int]) = {
+  private def getMetsLocation(key: Version[String, Int]): Result[MetsLocation] =
     adapterStore.get(key) match {
       case Left(err)                   => Left(err.e)
       case Right(Identified(_, entry)) => Right(entry)
     }
-  }
 
-  private def getFromMetsStore(metsData: MetsData) = {
-    temporaryCredentialsStore.get(
-      ObjectLocation(metsData.bucket, s"${metsData.path}/${metsData.file}"))
+  private def getMetsXml(metsLocation: MetsLocation): Result[MetsXml] =
+    metsXmlStore
+      .get(metsLocation.xmlLocation)
+      .flatMap { xml =>
```

Comment:
Same here: I can't find tests for this bit? it would be good to have a happy path test with multiple manifestations and a test where some of them fail at least. Could mean that this needs to be extracted into its own class to make test easier